contributors/devel/sig-architecture/validation-suites.md

```diff
@@ -0,0 +1,82 @@
+# Validation Suite Testing in Kubernetes
+
+A Kubernetes Validation suite is a subset of e2e tests that are used to
+validate correct functionality of an optional capability of Kubernetes. For example,
+Validation suites may be used to verify that optional storage or networking 
+capabilities are performing correctly.  They may also 
+be used to identify a group of tests that validate correct functionality on an
+optional platform such as Kuberenetes on Windows.  Kubernetes SIG groups are
+welcome to define Validation suites as needed.  Validation suites should not 
+be confused with [Conformance Tests](../sig-architecture/conformance-tests.md)
+which instead refers to the SIG Architecture core set of interoperable features that
+all conformant Kubernetes clusters must support. Instead, Validation suites 
+validate correct behavior on an optional or extension capability.  Over time a
+a Validation suite or portion of it may be promoted to become a 
+[Conformance Test](../sig-architecture/conformance-tests.md) if deemed necessary. 
+
+## Validation Suite Test Requirements
+
+Validation Test Suites may be used as deemed necessary by the associated Kubernetes
+SIG that is defining the Validation Suite.  Unlike Conformance Tests, Validation
+Suites may change over time with additions, deletions, and modifications whenever
```

Comment:
I find this description rather confusing, possibly because I simply disagree with it.

Surely we should have optional functionality which is none-the-less GA, and subject to all the same backwards compatibility guarantees as non-optional GA features?  And this admittedly probably bleeds into the profiles discussion.

If I as a user decide to use an optional capability (optional in the sense that it's allowable to have a cluster called Kubernetes even if it does not provide the capability, like remote attached storage), that is GA (in the sense that the API is fixed) I don't want the definition of how remote attached storage works to change from one release to the next?  Probably better discussed further on a call.

Reply:
@quinton-hoole-2 this takes into account how nodeconformance, csi test suite etc grew over time. note that even CRI is v1alpha2 (if i remember right). 

If you need guarantees, then that would be a conformance suite ... not a validation suite.

Reply:
@dims That may be true, but that's not what this document says.  

"A Kubernetes Validation suite is a subset of e2e tests that are used to
validate correct functionality of an optional capability of Kubernetes."

Reply:
@quinton-hoole-2 may be the missing piece is "for use with a specific release"? (or subject to change across release boundaries)

Reply:
Perhaps we have 3 separate axes that we're confusing in this doc:

1. optional vs non-optional
2. alpha vs beta vs GA
3. Kubernetes release number

a. The degree to which tests can change over time has to do with the axes 2 and 3.  If a test passes for a GA feature in release number x, then that same test should continue to pass, unchanged, essentially for all releases subsequent to x (due to backwards compatibility guarantees).
b. If release number x+n adds additional optional fields to an API, then a (logically) new test (to validate that those fields are correctly handled if present) for release x+n is required, and depending on whether the fields are alpha, beta or GA, the test may be required to pass for all releases subseqent to x+n.
c. The degree to which a test is required to pass in order for a cluster to be conformant has to do with axis 1.

Thoughts?

Reply:
Good point made above. I will address in the next version.

Reply:
@bradtopol I don't think this was ever addressed